### ISSUE TITLE
Can't connect to App. Have you connected the device?

### ISSUE THREAD

**Issue by Jnchi** [Nightbringer21/fridump#21]
```cmd
frida-ps -U
 PID  Name
----  --------------------------------------------------------
2510  Camera
2661  Files
2662  Filza
2666  Photos
1637  ScreenshotServicesService
2660  AppName :: <-- I want this one
2123  User Authentication
1663  AppPredictionWidget
 708  AppleIDAuthAgent
2250  AssetCacheLocatorService
 628  BlueTool
[ . . . TRUNCATED . . . ]

:: Real app name has been redacted
fridump.py -v -U AppName
Can't connect to App. Have you connected the device?
DEBUG:timeout was reached

fridump.py -v -U ReportCrash
Can't connect to App. Have you connected the device?
DEBUG:ambiguous name; it matches: ReportCrash (pid: 2507), ReportCrash (pid: 2287)

frida-trace -U 2660
Failed to attach: timeout was reached
```

iPhone: iPhone 6s Plus
Model: A1634
iOS: 11.2.6
Electra: 1.3.2

See also: https://github.com/frida/frida-python/issues/144, https://github.com/frida/frida-core/issues/163

**Comment by Jnchi** [comment]
```python
try:
    if USB:
        session = frida.get_usb_device().attach(APP_NAME)
```
Source: https://github.com/Nightbringer21/fridump/blob/master/fridump.py#L71

**Comment by Nightbringer21** [comment]
Can you provide a bit more information regarding the app that you are trying to attach to?

Primarily, does the app have any spaces in the name. If yes, you might need to enclose it within quotes.
E.g. fridump.py -v -U "App Name"

Second, I can see that frida-trace is not able to connect to the application as well. Are you sure that frida-server is running on the device?

**Comment by Jnchi** [comment]
Yes, the actual app name is a single word, e.g., "AppName". Also, `frida-server` was running fine on the device.

I was using Windows 10 on a laptop, but I tried from a MacBook instead, and I no longer had the issue. Feel free to close this.

**Comment by stirante** [comment]
I know it's a bit old issue, but I found a workaround for windows here https://github.com/frida/frida-python/issues/88

Just put `1` as a parameter for `get_usb_device()` and maybe base it on `enumerate_devices()`

**Comment by poldenais** [comment]
> Can you provide a bit more information regarding the app that you are trying to attach to?
> 
> Primarily, does the app have any spaces in the name. If yes, you might need to enclose it within quotes.
> E.g. fridump.py -v -U "App Name"
> 
> Second, I can see that frida-trace is not able to connect to the application as well. Are you sure that frida-server is running on the device?

@Nightbringer21 if an app name has two spaces in it does it matter. Like "App Test Name"?

I'm getting fridump: error: unrecognized arguments: Test Name”

it's like it doesn't like the second space?

**Comment by tonyarris** [comment]
I was getting the same error using a lightning cable. I switched to a classic USB cable and it ran fine.
